app/anti_virus_check.py

```diff
@@ -0,0 +1,136 @@
+import collections
+import time
+
+import requests
+from sdc.rabbit.exceptions import QuarantinableError, RetryableError, BadMessageError
+
+from app import create_and_wrap_logger
+from app import settings
+
+logger = create_and_wrap_logger(__name__)
+
+AVResult = collections.namedtuple('AVResult', 'safe ready scan_results')
+
+
+class AntiVirusCheck:
+    def __init__(self, tx_id):
+        self.bound_logger = logger.bind(tx_id=tx_id)
+
+    def send_for_av_scan(self, payload):
+        self.bound_logger.info("Sending for AV check", filename=payload.file_name)
+        data_id = self._send_for_anti_virus_check(payload.file_name, payload.decoded_contents)
+        self.bound_logger.info("Sent for A/V check", data_id=data_id)
+
+        attempts = 0
+        while attempts <= settings.ANTI_VIRUS_MAX_ATTEMPTS:
+            attempts += 1
+            results = self._get_anti_virus_result(data_id)
+            if not results.ready:
+                time.sleep(settings.ANTI_VIRUS_WAIT_TIME)
```

Comment:
This will block the queue consumer. Not much you can do as you don't have an async solution, but it may be worth adding a comment to highlight it for future devs.